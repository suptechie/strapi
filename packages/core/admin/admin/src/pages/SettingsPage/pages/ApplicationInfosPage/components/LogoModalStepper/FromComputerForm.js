import React, { useState, useRef } from 'react';
import styled from 'styled-components';
import { Box } from '@strapi/design-system/Box';
import { Flex } from '@strapi/design-system/Flex';
import { Icon } from '@strapi/design-system/Icon';
import { Typography } from '@strapi/design-system/Typography';
import { Button } from '@strapi/design-system/Button';
import PicturePlus from '@strapi/icons/PicturePlus';

const FileInput = styled(Box)`
  opacity: 0;
`;

const FromComputerForm = () => {
  const [dragOver, setDragOver] = useState(false);
  const inputRef = useRef(null);

  const handleDragEnter = () => setDragOver(true);
  const handleDragLeave = () => setDragOver(false);

  const handleClick = e => {
    e.preventDefault();
    inputRef.current.click();
  };

  const handleDrop = e => {
    const files = e?.dataTransfer?.files;
    console.log(files);
    handleDragLeave();
  };

  const handleChange = () => {
    const files = inputRef.current.files;
    console.log(files);
    handleDragLeave();
  };

  return (
    <form>
      <Box paddingLeft={2} paddingRight={2} paddingTop={6} paddingBottom={6}>
        {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
        <label>
          <Box
            paddingTop={9}
            paddingBottom={7}
            hasRadius
            justifyContent="center"
            background={dragOver ? 'primary100' : 'neutral100'}
            borderColor={dragOver ? 'primary500' : 'neutral300'}
            borderStyle="dashed"
            borderWidth="1px"
            position="relative"
            onDragEnter={handleDragEnter}
            onDragLeave={handleDragLeave}
            onDrop={handleDrop}
          >
            <Flex justifyContent="center">
              <Flex direction="column">
                <Icon
                  color="primary600"
                  width={`${60 / 16}rem`}
                  height={`${60 / 16}rem`}
                  as={PicturePlus}
                />

                <Box paddingTop={3} paddingBottom={5}>
                  <Typography variant="delta">Drag & Drop here or</Typography>
                </Box>

                <FileInput
                  accept=".jpg,.png,.svg"
                  cursor="pointer"
                  as="input"
                  position="absolute"
                  left={0}
                  right={0}
                  bottom={0}
                  top={0}
                  width="100%"
                  type="file"
                  name="files"
                  tabIndex={-1}
                  zIndex={1}
                  onChange={handleChange}
                  ref={inputRef}
                />

                <Button type="button" onClick={handleClick}>
                  Browse files
                </Button>

                <Box paddingTop={6}>
                  <Typography variant="pi" textColor="neutral600">
                    Max dimension: 750*750, Max size: TBC
                  </Typography>
                </Box>
              </Flex>
            </Flex>
          </Box>
        </label>
      </Box>
    </form>
  );
};

export default FromComputerForm;
