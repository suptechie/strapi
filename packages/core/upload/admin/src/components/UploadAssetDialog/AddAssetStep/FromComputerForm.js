/* eslint-disable jsx-a11y/label-has-associated-control */
import React, { useRef } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Box } from '@strapi/parts/Box';
import { Row } from '@strapi/parts/Row';
import { VisuallyHidden } from '@strapi/parts/VisuallyHidden';
import { H3 } from '@strapi/parts/Text';
import { ModalFooter } from '@strapi/parts/ModalLayout';
import { Button } from '@strapi/parts/Button';
import AddAssetIcon from '@strapi/icons/AddAsset';
import { useIntl } from 'react-intl';
import { getTrad } from '../../../utils';
import { typeFromMime } from '../../../utils/typeFromMime';
import { AssetSource } from '../../../constants';

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

const IconWrapper = styled.div`
  font-size: ${60 / 16}rem;

  svg path {
    fill: ${({ theme }) => theme.colors.primary600};
  }
`;

const MediaBox = styled(Box)`
  border-style: dashed;
`;

export const FromComputerForm = ({ onClose, onAddAsset }) => {
  const { formatMessage } = useIntl();
  const inputRef = useRef(null);

  const handleClick = e => {
    e.preventDefault();
    inputRef.current.click();
  };

  const handleChange = () => {
    const files = inputRef.current.files;
    const assets = [];

    for (let i = 0; i < files.length; i++) {
      const file = files.item(i);

      assets.push({
        source: AssetSource.Computer,
        type: typeFromMime(file.type),
        url: URL.createObjectURL(file),
        ext: file.name.split('.').pop(),
        mime: file.type,
      });
    }

    onAddAsset(assets);
  };

  return (
    <form>
      <Box paddingLeft={8} paddingRight={8} paddingTop={6} paddingBottom={6}>
        <label>
          <MediaBox
            paddingTop={11}
            paddingBottom={11}
            hasRadius
            justifyContent="center"
            borderColor="neutral300"
            background="neutral100"
          >
            <Row justifyContent="center">
              <Wrapper>
                <IconWrapper>
                  <AddAssetIcon aria-hidden />
                </IconWrapper>

                <Box paddingTop={3} paddingBottom={5}>
                  <H3 textColor="neutral600" as="span">
                    {formatMessage({
                      id: getTrad('input.label'),
                      defaultMessage: 'Drag & Drop here or',
                    })}
                  </H3>
                </Box>

                <Button type="button" onClick={handleClick}>
                  {formatMessage({
                    id: getTrad('input.button.label'),
                    defaultMessage: 'Browse files',
                  })}
                </Button>

                <VisuallyHidden>
                  <input
                    type="file"
                    multiple
                    name="files"
                    tabIndex={-1}
                    ref={inputRef}
                    onChange={handleChange}
                  />
                </VisuallyHidden>
              </Wrapper>
            </Row>
          </MediaBox>
        </label>
      </Box>

      <ModalFooter
        startActions={
          <Button onClick={onClose} variant="tertiary">
            {formatMessage({
              id: 'app.components.Button.cancel',
              defaultMessage: 'cancel',
            })}
          </Button>
        }
      />
    </form>
  );
};

FromComputerForm.propTypes = {
  onClose: PropTypes.func.isRequired,
  onAddAsset: PropTypes.func.isRequired,
};
