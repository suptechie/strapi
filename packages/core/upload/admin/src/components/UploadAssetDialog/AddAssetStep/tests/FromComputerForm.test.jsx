import React from 'react';

import { lightTheme, ThemeProvider } from '@strapi/design-system';
import { render as renderTL } from '@testing-library/react';
import { IntlProvider } from 'react-intl';

import { FromComputerForm } from '../FromComputerForm';

jest.mock('@strapi/admin/strapi-admin', () => ({
  ...jest.requireActual('@strapi/admin/strapi-admin'),
  getFetchClient: jest.fn().mockReturnValue({
    get: jest.fn(),
  }),
}));

describe('FromComputerForm', () => {
  it('snapshots the component', async () => {
    const { container } = renderTL(
      <IntlProvider locale="en" messages={{}}>
        <ThemeProvider theme={lightTheme}>
          <FromComputerForm onClose={jest.fn()} onAddAssets={jest.fn()} />
        </ThemeProvider>
      </IntlProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
