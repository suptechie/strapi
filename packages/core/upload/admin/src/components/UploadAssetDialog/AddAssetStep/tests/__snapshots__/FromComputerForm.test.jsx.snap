// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FromComputerForm snapshots the component 1`] = `
.c0 {
  padding-block-start: 24px;
  padding-inline-end: 40px;
  padding-block-end: 24px;
  padding-inline-start: 40px;
}

.c1 {
  background: #f6f6f9;
  padding-block-start: 64px;
  padding-block-end: 64px;
  border-radius: 4px;
  border-color: #c0c0cf;
  border: 1px solid #c0c0cf;
  position: relative;
}

.c7 {
  padding-block-start: 12px;
  padding-block-end: 20px;
}

.c9 {
  position: absolute;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  z-index: 1;
  width: 100%;
}

.c11 {
  position: relative;
}

.c12 {
  background: #4945ff;
  padding-block: 8px;
  padding-inline: 8px;
  padding-inline-end: 16px;
  padding-inline-start: 16px;
  border-radius: 4px;
  border-color: #4945ff;
  border: 1px solid #4945ff;
  cursor: pointer;
}

.c17 {
  background: #f6f6f9;
  padding-block-start: 16px;
  padding-inline-end: 20px;
  padding-block-end: 16px;
  padding-inline-start: 20px;
}

.c8 {
  font-weight: 500;
  font-size: 1.6rem;
  line-height: 1.25;
  color: #666687;
}

.c16 {
  font-size: 1.2rem;
  line-height: 1.33;
  font-weight: 600;
  color: currentcolor;
}

.c3 {
  align-items: center;
  display: flex;
  flex-direction: row;
  justify-content: center;
}

.c4 {
  align-items: center;
  display: flex;
  flex-direction: row;
}

.c13 {
  align-items: center;
  display: flex;
  flex-direction: row;
  gap: 8px;
}

.c19 {
  align-items: center;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.c21 {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c14 {
  position: relative;
  outline: none;
}

.c14[aria-disabled='true'] {
  pointer-events: none;
}

.c14:after {
  transition-property: all;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c14:focus-visible {
  outline: none;
}

.c14:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c15 {
  height: 3.2rem;
  color: #ffffff;
}

.c15 svg {
  height: 1.2rem;
}

.c15[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
  color: #666687;
}

.c15[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
  color: #666687;
}

.c15:hover {
  border: 1px solid #7b79ff;
  background: #7b79ff;
}

.c15:active {
  border: 1px solid #4945ff;
  background: #4945ff;
}

.c20 {
  height: 3.2rem;
  border: 1px solid #dcdce4;
  background: #ffffff;
  color: #32324d;
}

.c20 svg {
  height: 1.2rem;
}

.c20[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
  color: #666687;
}

.c20[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
  color: #666687;
}

.c20:hover {
  background-color: #f6f6f9;
}

.c20:active {
  background-color: #eaeaef;
}

.c18 {
  border-radius: 0 0 4px 4px;
  border-top: 1px solid #eaeaef;
}

.c5 {
  flex-direction: column;
}

.c6 {
  font-size: 6rem;
}

.c6 svg path {
  fill: #4945ff;
}

.c2 {
  border-style: dashed;
}

.c10 {
  opacity: 0;
  cursor: pointer;
}

<div>
  <form>
    <div
      class="c0"
    >
      <label>
        <div
          class="c1 c2"
          justifycontent="center"
        >
          <div
            class="c3"
          >
            <div
              class="c4 c5"
            >
              <div
                class="c6"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="3.2rem"
                  viewBox="0 0 32 32"
                  width="3.2rem"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16 3a13 13 0 1 0 13 13A13.016 13.016 0 0 0 16 3m5 14h-4v4a1 1 0 0 1-2 0v-4h-4a1 1 0 0 1 0-2h4v-4a1 1 0 0 1 2 0v4h4a1 1 0 0 1 0 2"
                  />
                </svg>
              </div>
              <div
                class="c7"
              >
                <span
                  class="c8"
                >
                  Drag & Drop here or
                </span>
              </div>
              <input
                class="c9 c10"
                multiple=""
                name="files"
                tabindex="-1"
                type="file"
              />
              <div
                class="c11"
              >
                <button
                  aria-disabled="false"
                  class="c12 c13 c14 c15"
                  type="button"
                >
                  <span
                    class="c16"
                  >
                    Browse files
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </label>
    </div>
    <div
      class="c17 c18"
    >
      <div
        class="c19"
      >
        <div
          class="c13"
        >
          <button
            aria-disabled="false"
            class="c12 c13 c14 c20"
            type="button"
          >
            <span
              class="c16"
            >
              cancel
            </span>
          </button>
        </div>
        <div
          class="c13"
        />
      </div>
    </div>
  </form>
  <span
    class="c21"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </span>
</div>
`;
